### PR TITLE
Np 45600 phisycally delete publication

### PR DESCRIPTION
**Pull request by truhacevkir**
Hard deleted publication can be fetched in the same way as soft deleted, etc. tobmstone/redirect in return.
Keeping only entityDescription and doi of hard deleted publication.
Not removing files from s3 bucket for now.

### CODE REVIEW

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/groovy/nva.publication.api.rootplugin.gradle [hunk 1]
```diff
@@ -6,6 +6,7 @@ plugins{
 
 
 repositories {
+    mavenLocal()
```
[the remark this answers is not in the document]
Comment (truhacevkir) [reply buildSrc/src/main/groovy/nva.publication.api.rootplugin.gradle]: removed

File: buildSrc/src/main/groovy/nva.publication.api.java-conventions.gradle [hunk 2]
```diff
@@ -33,6 +33,7 @@ java {
 }
 
 repositories {
+    mavenLocal()
```
[the remark this answers is not in the document]
Comment (truhacevkir) [reply buildSrc/src/main/groovy/nva.publication.api.java-conventions.gradle]: removed

**Review by brinxmat — APPROVED** [review]

File: publication-rest/src/main/java/no/unit/nva/publication/delete/DeletePublicationHandler.java [hunk 3]
```diff
@@ -62,44 +65,74 @@ protected Void processInput(Void input, RequestInfo requestInfo, Context context
...
-    }
-
     @Override
     protected Integer getSuccessStatusCode(Void input, Void output) {
         return HttpStatus.SC_ACCEPTED;
     }
 
+    private static void validateHardDeletionRequest(RequestInfo requestInfo, Publication publication)
+        throws UnauthorizedException {
+        if (isEditor(requestInfo, publication) || canEditOwnInstitutionResources(requestInfo)) {
```
Comment (joachimjorgensen) [inline publication-rest/src/main/java/no/unit/nva/publication/delete/DeletePublicationHandler.java:86]: A Curator shouldn't be allowed to hard delete a publication, if I recall correctly?

File: publication-rest/src/test/java/no/unit/nva/publication/delete/DeletePublicationHandlerTest.java [hunk 4]
```diff
@@ -470,7 +472,30 @@ void shouldPersistUnpublishRequestWhenDeletingPublishedPublication()
...
     }
 
+    @Test
+    void shouldDeleteUnpublishedPublicationWhenUserIsEditor()
+        throws ApiGatewayException, IOException {
+        var publication = createUnpublishedPublication();
+
+        var publisherUri = publication.getPublisher().getId();
+        var request = createHandlerRequest(publication.getIdentifier(), randomString(),
+                                           publisherUri, EDIT_OWN_INSTITUTION_RESOURCES);
```
Comment (joachimjorgensen) [inline publication-rest/src/test/java/no/unit/nva/publication/delete/DeletePublicationHandlerTest.java:482]: EDIT_OWN_INSTITUTION_RESOURCES is used to define a Curator, not an Editor.

**Pull request merged**